production/helm/loki/templates/gateway/deployment-gateway.yaml

```diff
@@ -29,6 +29,12 @@ spec:
         {{- toYaml . | nindent 8 }}
         {{- end }}
       labels:
+        {{- with .Values.gateway.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+        {{- with .Values.loki.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
```

Comment:
how does k8s handle if the same label is specified on both, and the template ends up with a duplicate? would it be better to use a template here and leverage helm's merging capabilities?

Reply:
Fair point. If two label keys are duplicated then the latter label wins. I don't quite see the value of using a template/merge here as the only choice is which label should get precedent, the one defined in the generic `loki` section or the more specific label in the service section.
Currently, the labels from the `loki` section get precedent. However, the behavior for annotations is the other way around and that actually makes more sense to me. 
I'll switch the order for the labels.